File: src/main/java/hu/frontrider/gearcraft/blocks/BlockBase.java
package hu.frontrider.gearcraft.blocks;

import hu.frontrider.gearcraft.GearCraft;
import hu.frontrider.gearcraft.registry.TierRegistry;
import net.minecraft.block.Block;

public abstract class BlockBase extends Block {
    protected TierRegistry.Tier tier;

    public BlockBase(TierRegistry.Tier tier, String blockName, String tag) {
        super(tier.material, tier.mapColor);
        String suffix = "_" + blockName;
        if (tag != null)
            suffix += "_" + tag;
        setRegistryName(GearCraft.MODID, tier.name + suffix);
        setUnlocalizedName(tier.name + suffix);
        this.tier = tier;
        TierRegistry.Tier.setBlock(this, tier);
    }

    public BlockBase(TierRegistry.Tier tier, String blockName) {
        this(tier, blockName, null);
    }
}

File: mod/src/main/java/hu/frontrider/gearcraft/blocks/BlockBase.kt
package hu.frontrider.gearcraft.blocks

import hu.frontrider.gearcraft.core.util.factory.BlockFactory
import net.minecraft.block.Block
import net.minecraft.block.SoundType
import net.minecraft.block.material.MapColor
import net.minecraft.block.material.Material

/**
 * contains my generic initializer code, and nothing else.
 * */
open class BlockBase(
        resistance: Float,
        tool: String,
        miningLevel: Int,
        hardness: Float,
        soundType: SoundType,
        material: Material,
        mapColor: MapColor
) : Block(material,mapColor)  {

    init{
        this.soundType = soundType
        this.setResistance(resistance)
        this.setHardness(hardness)
        this.setHarvestLevel(tool,miningLevel)
    }
}